src/lib/bInterface.js

```diff
@@ -312,35 +314,30 @@ export const getStats = function (web3, networkId){
 //
 ////////////////////////////////////////////////////////////////////////////////
 
-export function toNumber(bignum,web3) {
-  return Number(web3.utils.fromWei(bignum))
-}
+export const toNumber = (bignum) => Number(fromWei(bignum))
+export const gemToNumber = (bignum, userInfo) => Number(toUiDecimalPointFormat(bignum, userInfo.miscInfo.gemDecimals))
 
-function calcNewBorrowAndLPrice(userInfo,
-                                dEth,
-                                dDai,
-                                ilk,
-                                web3) {
-  dEth = toNumber(dEth,web3)
-  dDai = toNumber(dDai,web3)
-  const ethDeposit = toNumber(userInfo.bCdpInfo.ethDeposit,web3)
-  const daiDebt = toNumber(userInfo.bCdpInfo.daiDebt,web3)
+const calcNewBorrowAndLPrice = (userInfo, dEth, dDai, ilk, isGem) => {
+  const {gemDecimals} = userInfo.miscInfo
+  dEth = toNumber(dEth)
+  dDai = toNumber(dDai)
+  const deposit = isGem ? gemToNumber(userInfo.bCdpInfo.ethDeposit, userInfo) : toNumber(userInfo.bCdpInfo.ethDeposit)
```

Comment:
both should have same handling.
take gem decimals in both.
also is it ethDeposit or gemDeposit?